pkg/reconciler/v1alpha1/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -100,7 +106,7 @@ func (state PipelineRunState) GetNextTasks(candidateTasks map[string]v1alpha1.Pi
 		if _, ok := candidateTasks[t.PipelineTask.Name]; ok && t.TaskRun != nil {
 			status := t.TaskRun.Status.GetCondition(apis.ConditionSucceeded)
 			if status != nil && status.IsFalse() {
-				if !(t.TaskRun.IsCancelled() || status.Reason == "TaskRunCancelled") {
+				if !(t.TaskRun.IsCancelled() || status.Reason == "TaskRunCancelled" || status.Reason == ReasonConditionCheckFailed) {
```

Comment:
nit: for the future, we need to put `TaskRunCancelled` in a constant :stuck_out_tongue:

Reply:
Done!